client/packages/system/src/Vaccination/api/useVaccination.ts

```diff
@@ -174,7 +174,8 @@ const useInsert = ({
         vaccinationDate: Formatter.naiveDate(input.date ?? new Date()),
         comment: input.comment,
         notGivenReason: input.notGivenReason,
-        stockLineId: input.stockLine?.id,
+        stockLineId:
+          input.given && !isOtherFacility ? input.stockLine?.id : undefined,
```

Comment:
Small fix to ensure if you select a batch, then change to not given, it doesn't save the batch